geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java

```diff
@@ -540,13 +561,23 @@ public void addShadowPartitionedRegionForUserPR(PartitionedRegion userPR) {
         if (logger.isDebugEnabled()) {
           logger.debug("{}: Created queue region: {}", this, prQ);
         }
+        if ((prQ != null) && this.cleanQueues && sender.isPersistenceEnabled()) {
+          // now, clean up the shadowPR's buckets on this node (primary as well as
+          // secondary) for a fresh start
+          Set<BucketRegion> localBucketRegions = prQ.getDataStore().getAllLocalBucketRegions();
```

Comment:
I haven't looked at the complete code path...What if this member is joining a running cluster; with another member hosting primary queue (processing on-going events). And when a new member joins with clean  queue option; is the expectation is to remove local and remote events too or only local...The bucketRegion.clear() may be distributed...And may try to remove entries from the redundant queues...Do you have test for this scenario...

Reply:
Added Test case